src/components/heading/heading.component.tsx

```diff
@@ -20,6 +20,7 @@ import {
 } from "./heading.style";
 import useLocale from "../../hooks/__internal__/useLocale";
 
+type HeadingTypes = "h1";
```

Comment:
```suggestion
type HeadingTypes = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
```
As we are giving consumers control over setting the heading type, we should make ensure the typing of the prop reflects this